SourceryRuntime/Sources/AST/Type.swift

```diff
@@ -16,6 +17,24 @@ public typealias AttributeList = [String: [Attribute]]
     /// Imports that existed in the file that contained this type declaration
     public var imports: [Import] = []
 
+    // sourcery: skipEquality
+    /// Imports existed in all files containing this type and all its super classes/protocols
+    public var allImports: [Import] {
+        return Type.gatherAllImports(for: self).removingDuplicates()
+    }
+
+    private static func gatherAllImports(for type: Type) -> [Import] {
+        var allImports: [Import] = Array(type.imports)
+        let typesToScan = Array(type.implements.values)
+            + Array(type.inherits.values)
+            + Array(type.basedWithTypes.values)
+
+        typesToScan.forEach { (typeToScan) in
+            allImports.append(contentsOf: Type.gatherAllImports(for: typeToScan))
```

Comment:
I don't think there is a need for recursion - in `updateTypeRelationship` we already flatten types hierarchy, meaning each type will have all the types it inherits (or protocols it implements) directly or indirectly through inheritance

Reply:
you are right, it wasn't needed, fixed, thanks